File: chap04/demo-modify-mcmc_settings.R
library(cmdstanr)

d <- read.csv(file='input/data-salary.csv')
data <- list(N=nrow(d), X=d$X, Y=d$Y)
model <- cmdstan_model('model/model4-4.stan')

init_fun <- function(chain_id) {
  set.seed(chain_id)
  list(a=runif(1,30,50), b=runif(1,0,1), sigma=5)
}

fit <- model$sample(
  data=data, seed=123,
  init=init_fun,
  chains=3, iter_warmup=500, iter_sampling=500, thin=2,
  parallel_chains=3,
  save_warmup=TRUE
)

File: chap04/demo-modify-mcmc_settings.py
import pandas
import cmdstanpy

d = pandas.read_csv('input/data-salary.csv')
data = d.to_dict('list')
data.update({'N':len(d)})
model = cmdstanpy.CmdStanModel(stan_file='model/model4-4.stan')

def init_fun():
    return dict(a=40, b=1, sigma=5)

fit = model.sample(
    data=data, seed=123,
    inits=init_fun(),
    chains=3, iter_warmup=500, iter_sampling=500, thin=2,
    parallel_chains=3,
    save_warmup=True
)
